src/components/Header/Header.tsx

```diff
@@ -0,0 +1,45 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+
+interface Props {
+  onAdd: (title: string) => Promise<void>;
+  isLoading: boolean;
+  hasActiveTodos: boolean;
+}
+
+export const Header: React.FC<Props> = ({
+  onAdd,
+  isLoading,
+  hasActiveTodos,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    await onAdd(todoTitle);
+    setTodoTitle('');
+  };
+
+  return (
+    <header className="todoapp__header">
+      <button
+        type="button"
+        className={classNames('todoapp__toggle-all', {
+          active: hasActiveTodos,
+        })}
+        aria-label={' '}
+      />
+
+      <form onSubmit={handleSubmit}>
+        <input
+          type="text"
+          className="todoapp__new-todo"
+          placeholder="What needs to be done?"
+          value={todoTitle}
+          onChange={(event) => setTodoTitle(event.target.value)}
+          disabled={isLoading}
+        />
+      </form>
+    </header>
+  );
+};
```

Comment:
Short idea is to get rid of `isLoading` in props and add inner `isInputDisabled` state. `isInputDisabled` will be changing inside `handleSubmit` 

I got your idea with `!!tempTodo`, but I think with state it's more natural

```suggestion
export const Header: React.FC<Props> = ({
  onAdd,
  hasActiveTodos,
}) => {
  const [todoTitle, setTodoTitle] = useState('');
  const [isInputDisabled, setIsInputDisabled] = useState(false);

  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    setIsInputDisabled(true);
    await onAdd(todoTitle);
    setIsInputDisabled(false);
    setTodoTitle('');
  };

  return (
    <header className="todoapp__header">
      <button
        type="button"
        className={classNames('todoapp__toggle-all', {
          active: hasActiveTodos,
        })}
        aria-label={' '}
      />

      <form onSubmit={handleSubmit}>
        <input
          type="text"
          className="todoapp__new-todo"
          placeholder="What needs to be done?"
          value={todoTitle}
          onChange={(event) => setTodoTitle(event.target.value)}
          disabled={isInputDisabled}
        />
      </form>
    </header>
  );
};
```